
---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/"
)
```
# DURIAN <img src="man/figures/durianlogo.svg" align="right" alt="" width="120" />

DURIAN (Deconvolution and mUltitask-Regression-based ImputAtioN) is a hybrid, iterative method for imputation of drop-out in single-cell expression data.  Automatic recovery of missing single-cell reads is supervised by deconvoluted bulk expression data.

For examples of pipelines involving DURIAN, users should consult [the scripts](https://github.com/mkarikom/DURIAN_paper) used to generate the benchmarking and figures for our paper.

## Installation
### Dependencies
```{julia,cache=TRUE}
ENV["R_HOME"]="/usr/local/lib/R"
Pkg.add("RCall")
Pkg.build("RCall")
using RCall

Pkg.add(url="https://github.com/mkarikom/DistributedStwdLDA.jl.git")
```

```{r,cache=TRUE}
# Install the latest version and dependencies from github
library(devtools)
devtools::install_github("mkarikom/MuSiC@biobaseImport")
devtools::install_github("mkarikom/mtSCRABBLE")
install.packages("foreach")

Sys.setenv(JULIA_HOME="/usr/local/julia-1.7.2/bin")
install.packages("JuliaCall")
library(JuliaCall)
julia_setup(JULIA_HOME = Sys.getenv("JULIA_HOME"),verbose=TRUE,rebuild = TRUE,install=FALSE)
```

### Package
```{r,cache=TRUE}
install_github("mkarikom/DURIAN")
```

## Module Selection
<div>
DURIAN supports the integration of custom deconvolution modules, even modules written in other languages, such as Julia.  Module selection is made using the `deconv_method` argument to `DURIAN::run_durian` and the default is `deconv_method = "MuSiC"`.

Current deconvolution alternatives include:

-   fast, NNLS regression using [MuSiC (R)](https://github.com/xuranw/MuSiC)
-   distributed Monte Carlo inference using [dsLDA (Julia)](https://github.com/mkarikom/DistributedStwdLDA.jl)
</div>

## Getting Started
<div>
Check out basic usage of DURIAN running and visualizing imputation of Human single-cell data in `vignette("basic_usage")`.
</div>

##### Load the package and data
```{r setup,warning=FALSE,message=FALSE,cache=TRUE}
library(DURIAN)
data(c("C","B","pDataC"))
```

##### Preprocess the data
```{r, cache=TRUE}
library(Seurat)

seur = CreateSeuratObject(counts=C)
seur = NormalizeData(seur)
genes = VariableFeatures(FindVariableFeatures(seur, selection.method = "vst", nfeatures = 1500))
comgenes = intersect(genes,rownames(B))
C_prepro = subsetsc(scremoutlier(C),geneids=comgenes,return_obj=TRUE,nsd=3)
pDataC_prepro = pDataC[colnames(C_prepro),]
```

##### Run imputation on the single-cell data, using the bulk data for supervision.
```{r,warning=FALSE,message=FALSE,results='hide',cache = TRUE}
impresult_list=run_durian(
      scrabble_parameters = c(1,1e-6,1e-4),
      nEM = 5,
      scdata = C_prepro,
      metadata = pDataC_prepro,
      bulkdata = B,
      deconv_method = "MuSiC",
      nIter_outer = 3,
      nIter_inner = 3,
      nSDCIters = 50000,
      DECONVGENETHRESH=-0.01,
      SCRGENETHRESH=-0.01,
      outerStats = FALSE,
      durianEps=1e-3,
      saveImputationLog = FALSE,
      saveDeconvolutionLog = FALSE,
      saveImputedStep=FALSE)
  impresult = impresult_list[["C"]]
```

##### Construct a low-dimensional UMAP embedding from imputed and original data
```{r,warning=FALSE,message=FALSE,results='hide',cache=TRUE}
library(umap)
library(ggplot2)
library(reshape2)
umap_imputed = umap(t(impresult))
umap_orig = umap(t(as.matrix(C_prepro)))
imputed_df = cbind(umap_imputed$layout,pDataC_prepro)
imputed_df$status = "Imputed"
orig_df = cbind(umap_orig$layout,pDataC_prepro)
orig_df$status = "Unimputed"
df = rbind(imputed_df,orig_df)
colnames(df) = c("UMAP1","UMAP2","cellID","cellType","sampleID","status")
df$status = factor(df$status,levels=c("Unimputed","Imputed"))
```

##### Plot the original and imputed data
```{r,warning=FALSE,message=FALSE,results='hide', fig.width=7,fig.height=3,dev = "svg",fig.ext="svg",cache=TRUE}
ggplot(df,aes(x=UMAP1, y=UMAP2,color=cellType)) + 
      geom_point(size=1)+
      facet_grid(~status,scales="free") + theme_bw()
```


